crates/primitives/src/header.rs

```diff
@@ -775,7 +775,17 @@ impl SealedHeader {
         }
 
         // timestamp in past check
-        if self.header.is_timestamp_in_past(parent.timestamp) {
+        if chain_spec.is_optimism_mainnet() {
+            #[cfg(feature = "optimism")]
```

Comment:
```rust
if cfg!(feature = "optimism") && chain_spec.is_optimism_mainnet()  &&
...
```

Reply:
`is_bedrock_active_at_block` is op feature gated

Reply:
```rust
if cfg!(feature = "optimism") {
   #[cfg(feature = "optimism")] 
   if ... {
   }
}
```

maybe too much of a nit, but i just didnt want that check to happen every time, and let it be optimized at compilation